staging/src/k8s.io/apiserver/pkg/server/hooks.go

```diff
@@ -47,6 +50,12 @@ type PostStartHookContext struct {
 	StopCh <-chan struct{}
 }
 
+// PreShutdownHookContext provides information about this API server to a PreShutdownHookFunc
+type PreShutdownHookContext struct {
+	// LoopbackClientConfig is a config for a privileged loopback connection to the API server
+	LoopbackClientConfig *restclient.Config
```

Comment:
Is this connection still usable? If stopCh is closed, we stop the listener afaik. So we cannot use the loopback anymore.

Reply:
Probably not.  Since the apiserver is shutting down the best you could hope for is a race condition.  I'll remove